Simple Incremental/Assets/Scripts/PlayerMovementController.cs

```diff
@@ -0,0 +1,30 @@
+﻿using UnityEngine;
+
+[RequireComponent(typeof(Rigidbody2D))]
+public class PlayerMovementController : MonoBehaviour
+{
+    [SerializeField] string horizontalAxis = "Horizontal";
+    [SerializeField] float horizontalSpeed = 50f;
+    [Range(0, .5f)] [SerializeField] private float horizontalSmoothing = .05f;
+
+    float horizontalForce;
+    Rigidbody2D rigidBody;
+    Vector2 currentVelocity = Vector2.zero;
+
+    private void Awake()
+    {
+        rigidBody = GetComponent<Rigidbody2D>();
+    }
+
+    private void Update()
+    {
+        horizontalForce = Input.GetAxis(horizontalAxis);
+    }
+
+    private void FixedUpdate()
+    {
+        var target = new Vector2(horizontalForce * horizontalSpeed * 10f * Time.deltaTime, rigidBody.velocity.y);
```

Comment:
I'm not usually a fan of magic numbers but 10f was used here to keep the horizontalSpeed to a more sane number. Having a speed of 500 looked a bit strange.

Reply:
There are a few concerns here.

1. Time.deltaTime is based on framerate, while FixedUpdate is based on FixedRate. Time.deltaTime will vary based on framerate. The higher number of frames you have, the slower you will move.

2. GetAxis returns an input-smoothed version of the input (smoothed by a calculation from the sensitivity value between each frame). If you were previously moving left (value of -1), the smoothness increment was 0.05f, and you change your input to the right (value of +1), It would actually read -.95 on the first frame of changed movement, -9 on the next, -.85 ...... .9, .95, through ~40 frames until it reaches 1.0.

Also, consider caching your "target" variable so you're not recreating the variable and garbage collecting it every physics frame. Something like this:

Vector2 targetVelocity = Vector2.zero;

private void Update()
{
targetVelocity.x = Input.GetAxis(horizontalAxis) * speedMultiplier;
}

private void FixedUpdate()
{
targetVelocity.y = rigidbody2d.velocity.y;
rigidbody2d.velocity = targetVelocity;
}

Reply:
1. I originally was thinking of using Time.fixedDeltaTime but I found this in the doc so that's why I went with Time.deltaTime:

> For reading the delta time it is recommended to use Time.deltaTime instead because it automatically returns the right delta time if you are inside a FixedUpdate function or Update function.

 I tested the game with various target frame rates set and it appears to work correctly at all framerates I tried.

2. I didn't realize GetAxis was also applying smoothing to it so I will be changing this to GetAxisRaw. My hope with using SmoothDamp below was so we could control the smoothing more directly and use what we felt was good. In my tests using raw input it feels much snappier.

3. I can swap this out for a cached version if you would like but since Vector2 is a struct and just a local method variable it's allocated on the stack and won't be subject to garbage collection in this case.

Reply:
2. Try using the GetAxis (and not GetAxisRaw). Modify your horizontal axis in your input manager, try playing around with the Snap feature (which snaps the response back to 0 when there is no input), and the sensitivity. If you can't get a "feel good" smoothing playing around with those 2 values in the input manager, then a more tailored approach makes sense.

3. I had not idea! I just looked that up. That's very cool!

Reply:
For the magic number maybe you can just add a private variable in the class and call it Damping or Normalize than give a comment that explains what it does. Otherwise any coder that looks at your code will have no idea.